pkg/operator/ceph/cluster/controller.go

```diff
@@ -218,15 +222,21 @@ func add(mgr manager.Manager, r reconcile.Reconciler, context *clusterd.Context)
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileCephCluster) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
 	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
-	reconcileResponse, err := r.reconcile(request)
+	reconcileResponse, cephCluster, err := r.reconcile(request)
+
 	if err != nil {
 		logger.Errorf("failed to reconcile. %v", err)
+
+		if !kerrors.IsNotFound(err) {
```

Comment:
Hum actually, why would the error be `kerrors.IsNotFound`? Also, the error is wrapped in `reconcile()` this will never work as intended.

Reply:
Right `reconcile` is sending nil error in case of not found. Shall I check the name of the object if it an empty string then do not log events?

Reply:
I guess we don't need to worry about that because if cephcluster did not found then err will be nil and we will not log events.